tests/integration/security/authz/authz_test.go

```diff
@@ -0,0 +1,338 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"istio.io/istio/pkg/http/headers"
+	"istio.io/istio/pkg/test/echo/check"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/common/config"
+	"istio.io/istio/pkg/test/framework/components/echo/common/ports"
+	"istio.io/istio/pkg/test/framework/components/echo/echotest"
+	"istio.io/istio/pkg/test/framework/components/echo/match"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/tests/common/jwt"
+)
+
+const (
+	callMultiplier = 5
+)
+
+func TestAuthz_Principal(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.authorization.mtls-local",
+			"security.authorization.grpc-protocol",
+			"security.authorization.tcp").
+		Run(func(t framework.TestContext) {
+			// Allow ns1.A to call every service in ns1.
+			allowed := apps.Ns1.A
```

Comment:
bikeshedding a lot here, but should we embedd Ns1? 99% of tests shouldn't really care about namespace so they can then keep doing `apps.A`

Reply:
I don't disagree ... definitely seems like number of namespaces could be a config option when we build it. Let me think about it.